### PR TITLE
Gh-266: Added docs on matched vertex and corner case

### LINKED ISSUE #266: Add matched vertex docs
This should include what matchedVertex is and the strange behaviour that can happen in some edge cases, described in this [Accumulo integration test](https://github.com/gchq/Gaffer/blob/553412c300e8f538214a61479dc17bdfde46eb3d/store-implementation/accumulo-store/src/test/java/uk/gov/gchq/gaffer/accumulostore/integration/AccumuloMatchedVertexIT.java#L229).

As referenced in the Gaffer 2 Deprecation PR Review: https://github.com/gchq/Gaffer/pull/2564#pullrequestreview-887312620

### PR DESCRIPTION
**Pull request by t92549**
# Related Issue

- Resolve #266

### CODE REVIEW

File: docs/reference/glossary.md [hunk 1]
```diff
@@ -5,18 +5,19 @@ hide:
...
-| API         | Application Programming Interface. An API is for one or more services / systems to communicate with each other |
-| JSON        | JavaScript Object Notation is a text based format for representing structure data based on JavaScript object syntax|
-| Element     | The word is used to describe the combination of both an edge and entity |
-| Entity      | An entity represents a point in a graph e.g. a person |
-| Stores      | A Gaffer store represents the backing database responsbile for storing or facilitating access to a graph |
-| Operations  | An operation is an instruction / function that you send to the API to manipulate and query a graph |
-| Vertex      | A vertex refers to the field in an entity that describes its type |
+| Term              | Description                          |
+| :---------------- | :----------------------------------- |
+| Entity            | An entity represents a point in a graph |
```
Comment (tb06904) [inline docs/reference/glossary.md:10]: A glossary should really be in alphabetical order

File: docs/reference/operations-guide/get.md [hunk 2]
```diff
@@ -1240,6 +1240,119 @@ Gets elements related to provided seeds. [Javadoc](https://gchq.github.io/Gaffer
...
+            "directedType" : "EITHER"
+        } ]
+        }
+        ```
+
+    === "Python"
+
+        ``` python
+        g.GetElements(
+        input=[
```
Comment (tb06904) [inline docs/reference/operations-guide/get.md:1285]: Think the indentation is off a bit in these examples, line continuations should be 4 spaces for python and line wrap should only be needed if it is over 79 chars.

File: docs/reference/operations-guide/get.md [hunk 3]
```diff
@@ -1240,6 +1240,119 @@ Gets elements related to provided seeds. [Javadoc](https://gchq.github.io/Gaffer
         } ]
         ```
 
+??? warning "Example fetching *duplicate* edges in corner cases"
+
+    Get entities and edges by entity id 3 and edge id 2 to 3.
+    Here we see a corner case where strange behaviour occurs when using the Accumulo store.
+    This happens when you provide an EntitySeed which matches the destination of an Edge, and an EdgeSeed which matches the same Edge.
```
Comment (tb06904) [inline docs/reference/operations-guide/get.md]: Might want to have bits where you refer to classes/code in back ticks e.g. `EntitySeed` like how `matchedVertex` is

File: docs/reference/glossary.md (hunk 1 above)
Comment (t92549) [reply docs/reference/glossary.md:10]: They weren't before so I thought I would shuffle them around in order of relevance

File: docs/reference/operations-guide/get.md (hunk 2 above)
Comment (t92549) [reply docs/reference/operations-guide/get.md:1285]: I agree it looks funky but have copied exactly from another example. I think maybe another ticket should be made to clean up all the code snippet indentations and make them consistent. Perhaps under https://github.com/gchq/gaffer-doc/issues/427?

File: docs/reference/glossary.md (hunk 1 above)
Comment (tb06904) [reply docs/reference/glossary.md:10]: True I think it was just something that was missed when it was first implemented, the dictionary definition of a glossary is an alphabetical list though so we might want to do it if its just a quick fix

File: docs/reference/operations-guide/get.md (hunk 2 above)
Comment (tb06904) [reply docs/reference/operations-guide/get.md:1285]: Fair point think there is a few places across the docs, might be one for the tidy up ticket as you said

File: docs/reference/operations-guide/get.md [hunk 4]
```diff
@@ -1240,6 +1240,119 @@ Gets elements related to provided seeds. [Javadoc](https://gchq.github.io/Gaffer
         } ]
         ```
 
+??? warning "Example fetching *duplicate* edges in corner cases"
+
+    Get entities and edges by entity id 3 and edge id 2 to 3.
+    Here we see a corner case where strange behaviour occurs when using the Accumulo store.
+    This happens when you provide an `EntitySeed` which matches the destination of an edge, and an `EdgeSeed` which matches the same edge.
+    Due to how the `matchedVertex` filtering works within Gaffer's Accumulo iterator code, this duplication is not removed.
```
Comment (GCHQDeveloper314) [inline docs/reference/operations-guide/get.md:1249]: If removing the duplication is desirable, then maybe we should add something explaining how this could be done?

Comment (t92549) [reply docs/reference/operations-guide/get.md:1249]: Do you mean by passing the results through ToSet?

Comment (GCHQDeveloper314) [reply docs/reference/operations-guide/get.md:1249]: Potentially yes. Could just be a brief sentence saying that it's possible to remove the duplication inside Gaffer.

**Review by GCHQDeveloper314 — APPROVED** [review]
